connector/src/main/java/org/jboss/as/connector/subsystems/resourceadapters/ConnectionDefinitionAdd.java

```diff
@@ -50,27 +60,39 @@
     @Override
     protected void populateModel(ModelNode operation, ModelNode modelNode) throws OperationFailedException {
         for (AttributeDefinition attribute : CONNECTION_DEFINITIONS_NODE_ATTRIBUTE) {
-           attribute.validateAndSet(operation, modelNode);
+            attribute.validateAndSet(operation, modelNode);
         }
     }
 
     @Override
-    protected void performRuntime(OperationContext context, ModelNode operation, ModelNode recoveryEnvModel) throws OperationFailedException {
+    protected void performRuntime(OperationContext context, ModelNode operation, final Resource resource) throws OperationFailedException {
 
         final ModelNode address = operation.require(OP_ADDR);
         PathAddress path = PathAddress.pathAddress(address);
-        final String archiveName = path.getElement(path.size() -2).getValue();
+        final String jndiName = JNDINAME.resolveModelAttribute(context,operation).asString();
+        final String raName = path.getElement(path.size() - 2).getValue();
+        final String archiveOrModuleName;
+        ModelNode raModel = context.readResourceFromRoot(path.subAddress(0, path.size() - 1)).getModel();
```

Comment:
you really want this? this will give you whole model from root of resource tree up to passed path?
you probably want something like this:

ModelNode raModel = context.readResource(path.getParent()).getModel();
or even
ModelNode raModel = Resource.Tools.readModel(context.readResource(path.getParent()),<opitional-how-many-levels, default is all>);
if you need recursive model from that point